tensorflow/core/framework/numeric_types.h

```diff
@@ -46,7 +46,7 @@ struct bfloat16 {
   EIGEN_DEVICE_FUNC bfloat16() {}
 
   EIGEN_DEVICE_FUNC explicit bfloat16(const float v) {
-    if (isnan(v)) {
+    if (std::isnan(v)) {
```

Comment:
Would it be better to use `Eigen::numext::isnan` here, would that be a better pick for on-device code?
@hawkinsp @zheng-xq